libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/BaseGradleIT.kt

```diff
@@ -122,7 +122,7 @@ abstract class BaseGradleIT {
 
         // gradle wrapper version to wrapper directory
         private val gradleWrappers = hashMapOf<String, File>()
-        private const val MAX_DAEMON_RUNS = 30
+        private const val MAX_DAEMON_RUNS = 100
```

Comment:
I'm curious, why do we need to increase this? The existing infra should restart daemon when MAX_DAEMON_RUNS is reached.

Reply:
The test infra keeps track of the number of active daemons (`MAX_ACTIVE_GRADLE_PROCESSES=1`) and the number of times daemons are invoked (`MAX_DAEMON_RUNS=30`). I'm not sure why the second is necessary, but if I keep the old number, this PR--which adds more tests--will cross this threshold and start failing the last few tests. I added a comment below: 
```
if (DaemonRegistry.runCountForDaemon(version) >= MAX_DAEMON_RUNS) {
   // Warning: Stopping the Gradle daemon while the test suite has not finished running can break tests in weird ways.
   // TODO: Find a safe way to do this or consider deleting this code.
   stopDaemon(version, environmentVariables)
}
```

Increasing this threshold to ~40 would have fixed the issue, but the next time someone else runs into this issue, they would need to increase it again (after some significant time debugging if they're not aware of this code), so I increased it to 100 to avoid that trouble.

Reply:
SGTM